common/web/keyboard-processor/src/text/systemStores.ts

```diff
@@ -42,6 +42,11 @@ namespace com.keyman.text {
     }
 
     set(value: string) {
+      // No need to signal changes when things stay the same.
+      if(this._value == value) {
+        return;
+      }
+
```

Comment:
I think this might break some logic we are depending on. For example, if you want to override the layer switch for digits, you may have the following:

```
  + any(digit) > index(digit, 1) set(&layer = 'numeric')
```

Now, if we are already on the 'numeric' layer, it is still important that we make note of this "change", so that the `&newLayer` store is set when we call the `begin postkeystroke` entry point. That allows for the keyboard to test if the layer was "changed" or otherwise set by code in the `begin unicode` entry point, e.g. like this:

```
group(PostKeystroke) readonly
    if(&newLayer = "") > set(&layer = 'default')
```

So, would the `&newLayer` store be set in this instance, given `this.handler()` is not called? From my reading of the code, I don't think that it would be.

Reply:
Yeah, it took us a couple of iterations to get the logic solid on this. Caps Lock and layer switching is painfully complicated.